### PR TITLE
Implement TSTR Summary

### LINKED ISSUE #232: Train Synthetic Test Real Module
**What is your feature request related to?**
* Module request

**Please describe the module/problem**

It would be cool to have a module that implements the TSTR strategy for evaluating synthetic data. It should be possible to use classification and regression tasks for evaluation.

### PR DESCRIPTION
**Pull request by benHeid**
## Related Issues
closes #232 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos)
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] Did you update the CHANGELOG

## PR review
Anyone in the community is free to review the PR once the tests have passed.

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones to the PR so it can be classified

### CODE REVIEW

File: pywatts/summaries/train_synthetic_test_real.py [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+
+        if get_model is None:
+            self._get_model = self._get_regressor if task == TSTRTask.Regression else self._get_classifier
+        else:
+            self._get_model = get_model
+
+    @staticmethod
+    def _get_regressor(horizon, pred_horizon):
+        """
+        Default Regressor
```
Comment (marianturowski) [inline pywatts/summaries/train_synthetic_test_real.py:70]: Default regressor

File: pywatts/summaries/train_synthetic_test_real.py [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+        :param train_test_split: The share of data that should be used for training.
+        :type train_test_split: float
+        :param repetitions: The number of repetitions a model should be trained and evaluated on each data set.
+        :type repetitions: int
+        :param fit_kwargs: Key Word arguments for fitting the model.
+        :type fit_kwargs: Dict
+        :param get_model: A function that returns a model that implements a fit and a predict method.
+        :type get_model: Callable
+        :param n_targets: The number of classes for the classifaction task or the horizon for the regression task.
+        :type n_targets: int
```
Comment (marianturowski) [inline pywatts/summaries/train_synthetic_test_real.py:123]: descriptions of params task and metrics are missing

File: pywatts/summaries/train_synthetic_test_real.py [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+            return self._mape
+        elif metric_name == "f1":
+            return self._f1
+        elif metric_name == "accuracy":
+            return self._accuracy
+
+    def _get_regression_data(self, real, synthetic):
+        """
+        Get a data set for the regression task
+        """
```
Comment (marianturowski) [inline pywatts/summaries/train_synthetic_test_real.py:230]: Do you want to describe the params?

File: pywatts/summaries/train_synthetic_test_real.py [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+        X_train = synthetic[:split_index, :-self.n_targets]
+        y_train = synthetic[:split_index, -self.n_targets:]
+        X_test = real[split_index:, :-self.n_targets]
+        y_test = real[split_index:, -self.n_targets:]
+        return X_train, y_train, X_test, y_test
+
+    def _get_classification_data(self, real, synthetic, real_label, synthetic_label):
+        """
+        Get a data set for the classification task
+        """
```
Comment (marianturowski) [inline pywatts/summaries/train_synthetic_test_real.py:241]: Do you want to describe the params?

File: pywatts/summaries/train_synthetic_test_real.py [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+        X_train = synthetic[:split_index]
+        y_train = synthetic_label[:split_index]
+        X_test = real[split_index:]
+        y_test = real_label[split_index:]
+        return X_train, y_train, X_test, y_test
+
+    def _evaluate(self, train_x, train_y, test_x, test_y, n_targets, name):
+        """
+        Evaluate the synthetic data
+        """
```
Comment (marianturowski) [inline pywatts/summaries/train_synthetic_test_real.py:252]: Do you want to describe the params?

**Review by marianturowski — CHANGES_REQUESTED** [review]
Thank you for implementing the TSTR summary! Nice job!
I found some places where you might want to adapt the comments.
